### DISCUSSION TITLE
How to update these json fields in a smart way?

### DISCUSSION THREAD

**Discussion by hustcer** [nushell/nushell#8976] [Q&A] [answered]
Supporse we have a json file like this:

```json
{
    "name": "test",
    "dependencies": {
        "a": "1.0.0",
        "b": "1.0.0",
        "c": "1.0.0",
        "d": "1.0.0",
        "e": "1.0.0",
        "f": "1.0.0"
    }
}
```

How can we update a,b,c,d,e,f to 2.0.0 with as less code as possible without mention a,b,c... for a better extendability ?
A real world example: https://github.com/hustcer/nu-to-npm/blob/main/nu/bump-ver.nu

**Comment by amtoine** [comment]
@hustcer 
i got a solution, let me format it a bit :yum:

**Reply by amtoine** [reply, marked as answer]
let's say the data is in some `package.json` maybe :yum:
then you should be able to update it with
```bash
def bump [
    filename: path
    --version (-v): string
] {
    $filename | open | update dependencies {|it|
        $it.dependencies | transpose k v
        | update v { $version }
        | transpose -r | into record
    } | save --force $filename
}
```
and then
```bash
bump package.json --version 2.0.0
```

**Reply by hustcer** [reply]
Ah, how ingenious, thanks a lot.

**Reply by amtoine** [reply]
always happy to help :blush:

**Reply by amtoine** [reply]
i don't think we can reduce the `open` / `save` part that much :thinking: 
maybe the back-end-forth `transpose` but, yeah, the umbrella principle works quite well here :smirk:
